app/models/payment_provider_setup.rb

```diff
@@ -1,16 +1,35 @@
 require "flex_commerce_api/api_base"
+
+# @module FlexCommerce
 module FlexCommerce
   #
+  # @module PaymentProviderSetup
   # A flex commerce PaymentProvider model
   #
-  # This model provides access to the flex commerce PaymentProvider which is used to
-  # define a payment gateway and provide instructions to the front end application
-  # on how to set up the gateway in a browser (i.e. redirect to paypal etc..)
-  #
+  # This model is used to define a payment gateway and provide instructions
+  # to the front end application on how to set up the gateway in a browser 
+  # (i.e. redirect to paypal etc..)
   #
   #
-  class PaymentProviderSetup < FlexCommerceApi::ApiBase
-    belongs_to :payment_provider
+  module PaymentProviderSetup
 
+    class << self; attr_accessor :setup_type, :redirect_url; end
```

Comment:
This won't work for modules – you can have race conditions as this isn't thread safe.